src/Microsoft.AspNet.Authentication.Google/GoogleAuthenticationHandler.cs

```diff
@@ -30,45 +30,48 @@ public GoogleAuthenticationHandler(HttpClient httpClient)
             graphResponse.EnsureSuccessStatusCode();
             var text = await graphResponse.Content.ReadAsStringAsync();
             var user = JObject.Parse(text);
-
-            var context = new GoogleAuthenticatedContext(Context, Options, user, tokens);
-            var identity = new ClaimsIdentity(
-                Options.ClaimsIssuer,
-                ClaimsIdentity.DefaultNameClaimType,
-                ClaimsIdentity.DefaultRoleClaimType);
-
+            
+            var context = new GoogleAuthenticatedContext(Context, Options, Backchannel, user, tokens)
+            {
+                Properties = properties,
+                Principal = new ClaimsPrincipal(identity)
+            };
+            
             if (!string.IsNullOrEmpty(context.Id))
             {
                 identity.AddClaim(new Claim(ClaimTypes.NameIdentifier, context.Id,
                     ClaimValueTypes.String, Options.ClaimsIssuer));
             }
```

Comment:
@brentschmaltz ah, nice tweak, thanks :smile: 

As I'm pretty sure there are tons of places where we could use it, I'd suggest opening a new ticket to track it and implement it everywhere at the same time.